signwriting_animation/data/data_loader.py

```diff
@@ -0,0 +1,104 @@
+#dataloader
+import os
+import torch
+import pandas as pd
+from torch.utils.data import Dataset, DataLoader
+from torch.nn.utils.rnn import pad_sequence
+from pose_format import Pose
+
+class PosePredictionDataset(Dataset):
+    def __init__(self, data_dir, segmentation_csv, past_frames=40, future_frames=20, with_metadata=False):
+    """
+    Dataset for SignWriting skeletal pose animation prediction.
+    
+    Args:
+        csv_path (str or Path): Path to the CSV file containing segmentation information
+                                with columns: ['pose', 'start', 'end'].
+        data_dir (str or Path): Path to the directory containing raw .pose files.
+        past_frames (int): Number of frames used as input (past pose).
+        future_frames (int): Number of frames to be predicted (target pose).
+        dtype (np.dtype): Data type for the pose arrays. Default is np.float32.
+        with_metadata (bool): Whether to include pose metadata (e.g., FPS, width, height). Default: False.
+    """
+        self.data_dir = data_dir
+        self.with_metadata = with_metadata
+        self.past_frames = past_frames
+        self.future_frames = future_frames
+        self.samples = []
+
+        df = pd.read_csv(segmentation_csv)
+        for _, row in df.iterrows():
+            pose_file = row["pose"]
+            sample_path = os.path.join(data_dir, "raw_poses", pose_file)
+            if not os.path.exists(sample_path):
+                continue
+            self.samples.append({
+                "path": sample_path,
+                "id": pose_file,
+                "start": int(row["start"]),
+                "end": int(row["end"])
+            })
+
+    def __len__(self):
+        return len(self.samples)
+
+    def __getitem__(self, idx):
+        item = self.samples[idx]
+        with open(item["path"], "rb") as f:
+            pose = Pose.read(f.read())
+
+        data = pose.body.data.data
```

Comment:
you can do something like
```py
past_pose = pose.body[start:end]
future_pose = pose.body[end:end+self.future_frames]
```
instead of multiple slices - as in, you can slice directly the pose object